packages/playwright-core/src/server/electron/electron.ts

```diff
@@ -132,11 +133,12 @@ export class ElectronApplication extends SdkObject {
   }
 
   async close() {
-    const progressController = new ProgressController(serverSideCallMetadata(), this);
-    const closed = progressController.run(progress => helper.waitForEvent(progress, this, ElectronApplication.Events.Close).promise);
-    await this._browserContext.close({ reason: 'Application exited' });
-    this._nodeConnection.close();
-    await closed;
+    if (!this._startedClosing) {
+      this._startedClosing = true;
+      // This will call BrowserContext.setCustomCloseHandler.
+      await this._browserContext.close({ reason: 'Application exited' });
+    }
+    await this._appClosePromise;
```

Comment:
I think we should move this to line 93.5, otherwise `context.close()` does not actually wait for the process to exit. From what I see, our intent is that `app.close() === context.close() === gracefullyClose()`.

I also think we can remove `this._startedClosing` flag.

Reply:
Good catch.